test/test_autograd.py

```diff
@@ -7074,6 +7074,29 @@ def forward(self, dict_input):
         ):
             self.assertEqual(param.grad, checkpoint_param.grad)
 
+    def test_callback(self):
```

Comment:
How is this new test different from the one below? Is it just a simpler version that you used for debugging?